sql/moz-fx-data-shared-prod/firefox_ios_derived/app_store_retention_week_2_v1/query.sql

```diff
@@ -0,0 +1,48 @@
+WITH clients_retention AS (
+  SELECT
+    submission_date,
+    client_id,
+    sample_id,
+    days_seen_bits,
+  FROM
+    firefox_ios.baseline_clients_last_seen
+  WHERE
+    submission_date = @submission_date
+    AND normalized_channel = "release"
+),
+first_seen AS (
+  SELECT
+    client_id,
+    sample_id,
+    first_seen_date,
+  FROM
+    firefox_ios.baseline_clients_first_seen
+  -- Two weeks need to elapse before calculating the week 2 retention
+  WHERE
+    submission_date = DATE_SUB(@submission_date, INTERVAL 13 DAY)
```

Comment:
Strong opinion incoming: `first_seen` tables should not be filtered on `submission_date`. The grain of this table is `client_id`, and filtering on `submission_date` introduces possible error without much benefit (the table is very small, and joinable even at the scale of Firefox desktop).

The `submission_date` fields in any first_seen tables make them much more confusing to understand and use.

Reply:
so in general I agree here, unfortunately, due to how the underlying tables are set up this filter is required. Otherwise, we see the following error (when removing this filter):
```
Cannot query over table 'moz-fx-data-shared-prod.org_mozilla_ios_firefox_derived.baseline_clients_daily_v1' without a filter over column(s) 'submission_date' that can be used for partition elimination 
```